clang/lib/Frontend/CompilerInvocation.cpp

```diff
@@ -4198,6 +4198,9 @@ bool CompilerInvocation::ParseLangArgs(LangOptions &Opts, ArgList &Args,
   Opts.IncludeDefaultHeader = Args.hasArg(OPT_finclude_default_header);
   Opts.DeclareOpenCLBuiltins = Args.hasArg(OPT_fdeclare_opencl_builtins);
 
+  Opts.SYCLCUDACompat =
+      Args.hasArg(OPT_fsycl_cuda_compat, OPT_fno_sycl_cuda_compat, false);
```

Comment:
Use of `hasSYCLCudaCompatFlag()` would be preferred here. That means exposing it out of just `Clang.cpp` though.

Reply:
yes and sharing between the driver and frontend module.